src/UserHandler.cpp

```diff
@@ -0,0 +1,241 @@
+//
+// Created by idanto on 01/01/2020.
+//
+
+#include "../include/UserHandler.h"
+
+
+UserHandler::UserHandler(ConcurrentDataQueues& queues, ClientInventory& inventory) :
+        _queues(queues), _inventory(inventory), _actionFrameCommandMap(initActionFrameCommandMap()) {}
+
+
+void UserHandler::run() {
+    while (true) {
+        const short bufSize = 1024;
+        char buf[bufSize];
+        std::cin.getline(buf, bufSize);
+        std::string line(buf);
+        parseLine(line);
+    }
+}
+
+UserActionsEnum UserHandler::findCmd(std::string line) {
+    std::string cmd = line.substr(0, line.find(" "));
+    if (cmd == "login") {
+        return LOGIN;
+    }
+    else if (cmd == "join") {
+        return JOIN_GENRE;
+    }
+    else if (cmd == "exit") {
+        return EXIT_GENRE;
+    }
+    else if (cmd == "add") {
+        return ADD_BOOK;
+    }
+    else if (cmd == "borrow") {
+        return BORROW_BOOK;
+    }
+    else if (cmd == "return") {
+        return RETURN_BOOK;
+    }
+    else if (cmd == "status") {
+        return GENRE_BOOK_STATUS;
+    }
+    else { // cmd == logout. valid input guaranteed.
+        return LOGOUT;
+    }
+}
+
+void UserHandler::split(std::string& line, std::vector <std::string>& cmdParams, std::string delimiter) {
+    size_t start = 0U;
+    size_t end = line.find(delimiter);
+    while (end != std::string::npos) {
+        cmdParams.push_back(line.substr(start, end - start));
+        start = end + delimiter.length();
+        end = line.find(delimiter, start);
+    }
+    cmdParams.push_back(line.substr(start, line.size() - start + 1));
+}
+
+Frame UserHandler::receiveFrame() {
+    boost::unique_lock <boost::mutex> lock(_queues.mutexFromServer);
+    while (_queues.framesFromServer.empty()) _queues.condFromServer.wait(lock);
+    Frame ans = _queues.framesFromServer.front();
+    _queues.framesFromServer.pop();
+    return ans;
+}
+
+void UserHandler::sendFrame(Frame frame) {
+    boost::unique_lock <boost::mutex> lock(_queues.mutexToServer);
+    _queues.framesToServer.push(frame);
+    _queues.condToServer.notify_all();
+}
+
+void UserHandler::login(std::string& line) {
+    std::vector <std::string> cmdParams;
+    split(line, cmdParams, CMD_DELIMITER);
+    size_t portSeparator = cmdParams[1].find(":");
+    std::string host = cmdParams[1].substr(0, portSeparator);
+    std::string port = cmdParams[1].substr(portSeparator, cmdParams[1].size());
```

Comment:
Why not use split here too? with ":" as the delimiter. I think it's cleaner

Reply:
thought about it.. doesn't really save much lines, nor makes the code clearer. It's just more variables...